signer/src/storage/postgres.rs

```diff
@@ -190,6 +190,41 @@ impl PgStore {
 
         Ok(())
     }
+
+    async fn get_stacks_chain_tip(
+        &self,
+        bitcoin_chain_tip: &model::BitcoinBlockHash,
+    ) -> Result<Option<model::StacksBlockHash>, Error> {
+        sqlx::query_as!(
+            model::StacksBlock,
+            r#"
+            WITH confirmed_by_chain_tip AS (
+                SELECT
+                    stacks_blocks.block_hash
+                  , stacks_blocks.block_height
+                  , stacks_blocks.parent_hash
+                  , stacks_blocks.created_at
+                FROM sbtc_signer.stacks_blocks stacks_blocks
+                JOIN sbtc_signer.bitcoin_blocks bitcoin_blocks
+                    ON bitcoin_blocks.confirms @> ARRAY[stacks_blocks.block_hash]
+                WHERE bitcoin_blocks.block_hash = $1
+            )
+            SELECT
+                block_hash
+              , block_height
+              , parent_hash
+              , created_at
+            FROM confirmed_by_chain_tip
+            ORDER BY block_height DESC, block_hash DESC
+            LIMIT 1;
```

Comment:
Can't we use the order by and limit 1 without the CTE?

Reply:
Yes we can, and we will!

Reply:
And we have!